### PR TITLE
Added API allowing user to choose settings at runtime

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------

Test Steps
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: portable/Zynq.2019.3/ff_sddisk.c [hunk 1]
```diff
@@ -432,17 +432,19 @@ static CacheMemoryInfo_t * pucGetSDIOCacheMemory( BaseType_t xPartition )
...
+ * Initialize the SD Disk with configurable settings
+ * @param[in] uxMountFailIgnore ignore fails on mount, set to true when have systems where may not have initialized mount
+ * @param[in] uxDiskPartition the disk partition number to use
  */
-BaseType_t xDiskPartition = 0;
-
-FF_Disk_t * FF_SDDiskInit( const char * pcName )
+FF_Disk_t * FF_SDDiskInitWithSettings( const char * pcName,
+                                       BaseType_t xMountFailIgnore,
+                                       BaseType_t xDiskPartition )
```
Comment (phelter) [inline portable/Zynq.2019.3/ff_sddisk.c:447]: One slight change to allow for future proofing this.

Suggest putting all of the settings into a struct:

```
typedef struct FFInitSettings_s
{
BaseType_t xMountFailIgnore;
BaseType_t xDiskPartition;
} FFInitSettings_t;
```

And then change the API to:

```c
FF_Disk_t * FF_SDDiskInitWithSettings( const char * pcName,
                                       const FFInitSettings_t * settings)
```

That way you can append to the settings later on without changing the API - as long as the defaults are something reasonable (most likely candidate) the API doesn't have to change for new initialization configuration settings.

@AniruddhaKanhere - you are also missing adding the API to the header file.

**Review by phelter — CHANGES_REQUESTED** [review]

File: portable/Zynq.2019.3/ff_sddisk.c (hunk 1 above)
Comment (htibosch) [reply portable/Zynq.2019.3/ff_sddisk.c:447]: @phelter , yes a good idea to put it into a struct. The only obligation will be to clear all (unused) fields in the struct. Thanks, Hein

**Comment by phelter** [comment]
@AniruddhaKanhere and @htibosch -

Here's the modifications to the API and a fix for #50 that I'm now using.
https://github.com/FreeRTOS/Lab-Project-FreeRTOS-FAT/compare/main...phelter:Lab-Project-FreeRTOS-FAT:runtime-init-settings?expand=1

Feel free to use this or modify it as needed.

For all implementations but Zynq.2019.3 - uses original implementation - both WithSettings version and not do the same thing.  For the Zynq.2019.3 version - you have full control.

**Comment by AniruddhaKanhere** [comment]
Closing this in favor of: https://github.com/FreeRTOS/Lab-Project-FreeRTOS-FAT/pull/51

**Pull request closed without merging**